### DISCUSSION TITLE
unknown function "bpf_seq_write"

### DISCUSSION THREAD

**Discussion by Jianchen-Shan** [bpftrace/bpftrace#2786] [General]
Hello I am new to bpftrace and I am encountering an issue when using it after I customized my kernel with a patch found here: 

https://lore.kernel.org/bpf/CA+khW7i460ey-UFzpMSJ8AP9QeD8ufa4FzLA4PQckNP00ShQSw@mail.gmail.com/T/#mfcb3f1cf328c1c4a1c1740b4c325f9f34f916f34

As you can see from the output below, the examples show that one tool (e.g., vfsstat) can run correctly under the bcc directory, but they would fail under bpftrace/tools with an error complaining about an unknown function. Can anyone give me some insights?

I don't think the customized kernel is an issue since the BCC tools work well. I built bpftrace from the source as shown in the installation instructions on the ubuntu machine. I appreciate your help.

ubuntu@myvm:~/bpftrace/bcc/tools$ sudo ./vfsstat.py 
TIME         READ/s  WRITE/s  FSYNC/s   OPEN/s CREATE/s
00:15:57:      1330        3        0       19        0
00:15:58:        34        3        0       16        0
00:15:59:        70        3        0       19        0

ubuntu@myvm:~/bpftrace/bcc/libbpf-tools$ sudo ./vfsstat 
TIME         READ/s  WRITE/s  FSYNC/s   OPEN/s CREATE/s
00:16:47:        40        5        0       17        0
00:16:48:        71        4        0       19        0

ubuntu@myvm:~/bpftrace/tools$ sudo ./vfsstat.bt -v
INFO: node count: 31
Attaching 8 probes...

Error log: 
0: R1=ctx(off=0,imm=0) R10=fp0
0: (b7) r6 = 0                        ; R6_w=0
1: (7b) *(u64 *)(r10 -16) = r6        ; R6_w=P0 R10=fp0 fp-16_w=00000000
2: (18) r1 = 0xffff93d8ef573a00       ; R1_w=map_ptr(off=0,ks=0,vs=0,imm=0)
4: (bf) r2 = r10                      ; R2_w=fp0 R10=fp0
5: (07) r2 += -16                     ; R2_w=fp-16
6: (b7) r3 = 8                        ; R3_w=8
7: (b7) r4 = 0                        ; R4_w=0
8: (85) call bpf_seq_write#130
unknown func bpf_seq_write#130
processed 8 insns (limit 1000000) max_states_per_insn 0 total_states 0 peak_states 0 mark_read 0

ERROR: Error loading program: BEGIN

**Comment by danobi** [comment]
closing to discuss in issue instead
